src/app/technical-record/technical-record.component.ts

```diff
@@ -1,116 +1,116 @@
-
-    const editedRecord: TechRecord = this.allowedValues.mapControlValuesToDataValues(
-      JSON.parse(JSON.stringify(value))
-    );
-    const mergedRecord: TechRecord = { ...this.activeRecord, ...editedRecord };
-
-    const dialogRef = this.dialog.open(AdrReasonModalComponent, {
-      width: '600px',
-      data: {
-        context: 'Enter reason for changing technical record'
-      }
-    });
-
-    dialogRef.afterClosed().subscribe((result) => {
-      if (result && result.isSave) {
-        mergedRecord.reasonForCreation = result.data;
-        this.submitTechRecord.emit(mergedRecord);
-      }
-    });
-    // }
-  }
-}
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output
+} from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { MatDialog } from '@angular/material/dialog';
+
+import { AdrReasonModalComponent } from '@app/shared/adr-reason-modal/adr-reason-modal.component';
+import { TechRecordHelpersService } from '@app/technical-record/tech-record-helpers.service';
+
+import { TechnicalRecordValuesMapper } from './technical-record.mapper';
+import { TechRecord } from './../models/tech-record.model';
+import { MetaData } from '@app/models/meta-data';
+import { VehicleTechRecordModel } from '@app/models/vehicle-tech-record.model';
+import { TestResultModel } from '@app/models/test-result.model';
+import { VIEW_STATE } from '@app/app.enums';
+
+@Component({
+  selector: 'vtm-technical-record',
+  templateUrl: './technical-record.component.html',
+  styleUrls: ['../app.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class TechnicalRecordComponent implements OnInit {
+  showAdrDetails: boolean;
+  adrDisplayParams: { [key: string]: boolean };
+  techRecord: FormGroup;
+
+  allOpened = false;
+  panels: { panel: string; isOpened: boolean }[] = [
+    { panel: 'panel1', isOpened: false },
+    { panel: 'panel2', isOpened: false },
+    { panel: 'panel3', isOpened: false },
+    { panel: 'panel4', isOpened: false },
+    { panel: 'panel5', isOpened: false },
+    { panel: 'panel6', isOpened: false },
+    { panel: 'panel7', isOpened: false },
+    { panel: 'panel8', isOpened: false },
+    { panel: 'panel9', isOpened: false },
+    { panel: 'panel10', isOpened: false }
+  ];
+
+  @Input() vehicleTechRecord: VehicleTechRecordModel;
+  @Input() activeRecord: TechRecord;
+  @Input() metaData: MetaData;
+  @Input() editState: VIEW_STATE;
+  @Input() testResultJson: TestResultModel[];
+  @Output() submitTechRecord = new EventEmitter<TechRecord>();
+  @Output() changeViewState = new EventEmitter<VIEW_STATE>();
+
+  constructor(
+    private dialog: MatDialog,
+    private allowedValues: TechnicalRecordValuesMapper,
+    public techRecHelpers: TechRecordHelpersService
+  ) {}
+
+  ngOnInit() {
+    this.techRecord = new FormGroup({});
+    this.adrDisplayParams = { showAdrDetails: !!this.activeRecord.adrDetails };
+  }
+
+  togglePanel() {
```

Comment:
Can find the unit tests for that, this needs to be added if we don't have them please.

Reply:
will be resolved in 10163 as part of accordion refactor was code was done prior to ADR if needed. Reference ticket: https://jira.dvsacloud.uk/browse/CVSB-15024